receiver/httpcheckreceiver/metadata.yaml

```diff
@@ -54,3 +54,12 @@ metrics:
       monotonic: false
     unit: "{error}"
     attributes: [http.url, error.message]
+  httpcheck.substring:
+    description: Reports 1 if the HTTP response body matches the `substring` configuration item, 0 otherwise.
+    enabled: false
+    sum:
+      value_type: int
+      aggregation_temporality: cumulative
+      monotonic: false
+    unit: 1
+    attributes: [http.url]
```

Comment:
Shouldn't we add the body match string (the value of `substring`) as an attribute of this metric?

Let's imagine the following scenario: I have a web page that reports status of several things, and I want to use the HTTP Check receiver to check the status of those things. For example, it's a web page that has the following content:

```html
Component 1: OK
Component 2: Failed
Component 3: OK
```

My HTTP Check receiver configuration would then be:

```yaml
receivers:
  httpcheck:
    targets:
    - endpoint: http://my-web.page
      substring: "Component 1: OK"
    - endpoint: http://my-web.page
      substring: "Component 2: OK"
    - endpoint: http://my-web.page
      substring: "Component 3: OK"
```

With the proposed design of the `httpcheck.substring` metric, I would get three data points with the same name and attributes and couldn't differentiate between them:

```
httpcheck.substring{http.url="http://my-web.page"} 1
httpcheck.substring{http.url="http://my-web.page"} 0
httpcheck.substring{http.url="http://my-web.page"} 1
```

Another way to fix this might be to add a `name` property to the `target` definition, that could be used as an attribute.

Not sure if this example isn't stretching what the HTTP Check receiver is design to do though. @codeboten any thoughts?

Reply:
I agree with your point and I propose an optional `name` attribute. The default value would be an empty string, in which case the name attribute would be omitted.

Reply:
Thanks for your suggestion @atoulme! I have a doubt about how to implement the optional functionality. Do you mean add `name` attribute in targetConfig(Default is empty) and add `name` metric attribute?
``` Go
// receiver/httpcheckreceiver/config.go
type targetConfig struct {
	confighttp.HTTPClientSettings `mapstructure:",squash"`
	Method                        string `mapstructure:"method"`
	Body                          string `mapstructure:"body"`
	Name 	                      string `mapstructure:"name"`
}
```
``` Yaml
# receiver/httpcheckreceiver/metadata.yaml
attributes:
  http.url:
    description: Full HTTP request URL.
    type: string
  name:
    description: Body exact matches string.
    type: string
```

Reply:
yes

Reply:
Done. Would it be possible for help to review it again when you have some free time @atoulme and @astencel-sumo? Thanks!

Reply:
Marking this conversation as Resolved, as it is resolved from my perspective.